### PR TITLE
analyzer: Have exited processes read all data

### LINKED ISSUE #331: Analyzers that exit clean without reading input until EOF
@nwt's comment from https://github.com/brimdata/zui/issues/2926#issuecomment-1847706468:

> What's happening here is that Suricata is exiting cleanly (i.e., with status 0) without reading its standard input to EOF. When Suricata exits, io.Copy in the goroutine created by analyzer.runProcesses returns because writes to the pipe connected to its standard input start failing. The goroutine then closes Zeek's standard input, resulting in this error.

That Zui issue improved how Brimcap catches and presents this kind of error, but @nwt feels there's still improvements that can be made on the Brimcap side. A current quote:

> Right now brimcap assumes that analyzers that exit with status zero will read their standard input until EOF. I’m not sure that’s a good assumption.

### PR DESCRIPTION
**Pull request by mattnibs**
This commit changes the behavior for analyzer processes so that processes that have successfully exited without reading all the data will continue to consume data from the byte stream insteading of returning an error and putting a stop to the copy goroutine.

Closes #331

### CODE REVIEW

**Comment by philrz** [comment]
## tl;dr

This is a functional 👍 for me!

## Details

I pointed my Zui at the Brimcap commit from this branch and tested it out with the pcaps I mentioned in #331 comments and saw the improvements expected in both cases.

1. For the one in https://github.com/brimdata/brimcap/issues/331#issuecomment-1907185603, the error changes from this incorrect one:
 
image

to this accurate one:

image

and that latter error message will go away entirely when the Zeek v6.0.3-based artifact is in use since that has support for this link type.

2. For the one in https://github.com/brimdata/brimcap/issues/331#issuecomment-1907207180, it goes from showing this incorrect error message:

image

to a successful import:

image

i.e., Suricata quietly refused to do anything with this pcap since it doesn't support the link layer protocol, but since it returned an exit code of 0 the Zui user gets only Zeek events. This seems like a fine place to be while we wait to see if the Suricata people ever catch up and address the existing issues.

File: cmd/brimcap/ztests/analyze-process-readall.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    data: |
+      for i in {1..10000}; do
+        echo "{x: $i}"
+      done
+  - name: noread.sh
+    data: |
+      exit 0
+  - name: readall.sh
+    data: |
+      cat - > readall.zson
```
Comment (nwt) [inline cmd/brimcap/ztests/analyze-process-readall.yaml:31]: Feeding non-pcap data to `brimcap analyze` is surprising but fine I suppose.

But maybe shorten this?

```suggestion
script: |
  seq 10000 | brimcap analyze -config=config.yaml - | zq -z 'count()' -

inputs:
  - name: config.yaml
    data: |
      analyzers:
        - cmd: 'true'
          name: noread
        - cmd: bash
          args: [-c, 'cat > readall.zson']
          name: readall
```

**Review by nwt — APPROVED** [review]
